daemon/network.go

```diff
@@ -31,37 +31,48 @@ func (daemon *Daemon) NetworkControllerEnabled() bool {
 
 // FindNetwork function finds a network for a given string that can represent network name or id
 func (daemon *Daemon) FindNetwork(idName string) (libnetwork.Network, error) {
-	// Find by Name
-	n, err := daemon.GetNetworkByName(idName)
-	if err != nil && !isNoSuchNetworkError(err) {
+	// 1. match via full ID.
+	n, err := daemon.GetNetworkByID(idName)
+	if err == nil {
+		return n, nil
+	}
+	if !isNoSuchNetworkError(err) {
```

Comment:
Minor: this could be written as:

```go
if err == nil || !isNoSuchNetworkError(err) {
    return n, err
}
```

same below

Reply:
Yes! I like that 👍